include/asm/fstack.hpp

```diff
@@ -23,16 +24,18 @@ struct FileStackNode {
 	uint32_t ID; // Set only if referenced: ID within the object file, -1 if not output yet
 
 	enum FileStackNodeType type;
-};
-
-struct FileStackReptNode { // NODE_REPT
-	struct FileStackNode node;
-	std::vector<uint32_t> iters; // REPT iteration counts since last named node, in reverse depth order
-};
-
-struct FileStackNamedNode { // NODE_FILE, NODE_MACRO
-	struct FileStackNode node;
-	std::string name; // File name for files, file::macro name for macros
+	std::variant<
```

Comment:
What this loses is that each node is now `union`-like, i.e. the largest of the two sub-types. Since that only matters if each node is allocated manually, this is probably a good tradeoff.

Reply:
Also the vector and string are probably about the same size (pointer, capacity, length).